third_party/intel/lib/TritonIntelGPUTransforms/MatchTargetSize.cpp

```diff
@@ -137,6 +137,74 @@ class MatchTargetSizePass
 
     MLIRContext *ctx = &getContext();
     ModuleOp m = getOperation();
+    // this is ad-hoc for Q SLM
+    char *env = std::getenv("TRITON_INTEL_ENABLE_SLM");
+    const bool enableSLM = env ? (bool)std::atoi(env) : false;
+    if (enableSLM) {
+      dotWithSLMOperands.clear();
+      tt::LoadOp load;
+      m.walk([&](tt::LoadOp op) {
+        load = op;
+        return WalkResult::interrupt();
+      });
```

Comment:
Can something like below be done?
```suggestion
      tt::LoadOp load = m.getBody()->getOps<tt::LoadOp>().first;
```